etna/transforms/timestamp/fourier.py

```diff
@@ -20,17 +21,36 @@
 class FourierTransform(IrreversibleTransform):
     """Adds fourier features to the dataset.
 
+    Transform can work with two types of timestamp data: numeric and datetime.
+
+    Transform can accept timestamp data in two forms:
+
+    - As index. In this case the dataset index is used to compute features. The features will be the same for each segment.
+
+    - As external column. In this case for each segment its ``in_column`` will be used to compute features.
+      It is expected that for each segment we have the same type of timestamp data and for datetime type there is only one frequency.
+
+    If we are working with external column, there is a difference in handling numeric and datetime data:
+
+    - Numeric data can have missing values at any place.
+
+    - Datetime data could have missing values only at the beginning of each segment.
+
     Notes
     -----
-    To understand how transform works we recommend:
-    `Fourier series <https://otexts.com/fpp2/useful-predictors.html#fourier-series>`_.
+    To understand how transform works we recommend reading:
+    `Fourier series <https://otexts.com/fpp3/useful-predictors.html#fourier-series>`_.
 
-    * Parameter ``period`` is responsible for the seasonality we want to capture.
-    * Parameters ``order`` and ``mods`` define which harmonics will be used.
+    If we already have a numeric data then for a mode $m$ with a period $p$ we have:
 
-    Parameter ``order`` is a more user-friendly version of ``mods``.
-    For example, ``order=2`` can be represented as ``mods=[1, 2, 3, 4]`` if ``period`` > 4 and
-    as ``mods=[1, 2, 3]`` if 3 <= ``period`` <= 4.
+    .. math::
+        & k = \\left \\lfloor \\frac{m}{2} \\right \\rfloor
+        \\\\
+        & f_{m, i} = \\sin \\left( \\frac{2 \\pi k i}{p} + \\frac{\\pi}{2} (m \\mod 2) \\right)
+
+    If we have datetime data, then it first should be transformed into numeric.
+    During fitting the transform saves frequency and some datetime timestamp as a reference point.
```

Comment:
datetime timestamp ?

Reply:
I make a distinction between datetime and timestamp. Timestamp is a value that points to some point in a time series. It could be integer or datetime.

Reply:
I don't think that is clear from the context

Reply:
I rewrote this docstring a little.